### PR TITLE
Add some "paranoid" tests for new URL encoding algorithm + forbid 0x7f

### PR DESCRIPTION
**Pull request by snazy**
... follow-up of #9282

### CODE REVIEW

File: api/model/src/testUriCompliance/java/org/projectnessie/model/TestElementsEncoding.java [hunk 1]
```diff
@@ -495,4 +508,38 @@ static List<List<String>> randomRoundTrips() {
     }
     return r;
   }
+
+  @ParameterizedTest
+  @MethodSource
+  void allCharacters(String element) {
```
Comment (snazy) [inline api/model/src/testUriCompliance/java/org/projectnessie/model/TestElementsEncoding.java:514]: A very "paranoid" test that exercises all possible `char`s starting with ASCII 32.

File: api/model/src/testUriCompliance/java/org/projectnessie/model/TestElementsEncoding.java [hunk 2]
```diff
@@ -495,4 +509,38 @@ static List<List<String>> randomRoundTrips() {
...
+  }
+
+  static Stream<String> allCharacters() {
+    int split = 32;
+    return IntStream.rangeClosed(32 / split, 0xffff / split)
+        .mapToObj(
+            i -> {
+              char c = (char) (i * split);
+              StringBuilder sb = new StringBuilder(split);
+              for (int i1 = 0; i1 < 8; i1++) {
```
Comment (dimas-b) [inline api/model/src/testUriCompliance/java/org/projectnessie/model/TestElementsEncoding.java:540]: Each iteration span includes 32 chars, but we only return the first 8?

Comment (snazy) [reply api/model/src/testUriCompliance/java/org/projectnessie/model/TestElementsEncoding.java]: Argh - left-over.

However, 0x7f triggered a test failure. Just decided to forbid `DEL`

File: api/model/src/testUriCompliance/java/org/projectnessie/model/TestElementsEncoding.java [hunk 3]
```diff
@@ -495,4 +509,41 @@ static List<List<String>> randomRoundTrips() {
...
+
+  static Stream<String> allCharacters() {
+    int split = 32;
+    return IntStream.rangeClosed(32 / split, 0xffff / split)
+        .mapToObj(
+            i -> {
+              char c = (char) (i * split);
+              StringBuilder sb = new StringBuilder(split);
+              for (int i1 = 0; i1 < split; i1++) {
+                if (c != 0x7f) {
```
Comment (dimas-b) [inline api/model/src/testUriCompliance/java/org/projectnessie/model/TestElementsEncoding.java:541]: How does `DEL` fail?

Comment (dimas-b) [reply api/model/src/testUriCompliance/java/org/projectnessie/model/TestElementsEncoding.java:541]: New Servlet spec violation?

**Review by dimas-b — APPROVED** [review]

File: api/model/src/testUriCompliance/java/org/projectnessie/model/TestElementsEncoding.java (hunk 2 above)
Comment (dimas-b) [reply api/model/src/testUriCompliance/java/org/projectnessie/model/TestElementsEncoding.java]: Forbidding `DEL` SGTM

**Pull request merged**